synadm/api.py

```diff
@@ -556,6 +556,7 @@ def user_modify(self, user_id, password, display_name, threepid,
             data.update({"deactivated": True})
         if deactivation == "activate":
             data.update({"deactivated": False})
+        data.update({"user_type": user_type})
```

Comment:
What if it's None? What does the API respond then? Valid? Was that the M_UNKNOWN response you were referring to?

Reply:
Yes, `user_type` can be `null` in the JSON in which case the field gets cleared (which is basically the “default” option).

Reply:
Basically that's ok but the bad thing is that if any other property of a user gets changed, the user_type is always cleared. Do we have this behavior somewhere else already by any chance?

Reply:
Now that you mention it, it seems the activate/deactivate thing works just like that 😁 iʼll check tomorrow as itʼs bedtime for me now.

Reply:
I was right; the activate/deactivate feature works just like this. If i don’t add `--user-type` nor `--clear-user-type` to the command line, it won’t get updated:

```
$ synadm user modify --password abcdef @test:matrix.local
Current user account settings:
{"name": "@test:matrix.local "is_guest": 0, "admin": false, "consent_version": null, "consent_ts": null, "consent_server_notice_sent": null, "appservice_id": null, "creation_ts": 1457796444, "user_type": null, "deactivated": false, "shadow_banned": false, "displayname": "Test Account", "avatar_url": null, "threepids": [], "external_ids": [], "erased": false}
User account settings to be modified:
Password will be set as provided on command line.
Are you sure you want to modify user? (y/N):
```

Reply:
Just read the api docs: https://matrix-org.github.io/synapse/develop/admin_api/user_admin_api.html#create-or-modify-account

user_type is optional, I think you just need to make sure it isnt included at all when both of the 2 new options are not given on cli. So nothing gets cleared 🎉 Currently you always include it in the data sent to the endpoint.

Reply:
@gergelypolonkai I've just added the missing condition as suggested in my previous post. I've just edited your feature branch. I hope that's ok.

Reply:
This change was necessary, otherwise this would happen as with your original code:
```
(synadm38) ~/git/gergelypolonkai_synadm (user-type-edit) $ synadm user modify --display-name "TESTUSER1" testuser1
Current user account settings:
admin: 0
appservice_id: null
avatar_url: null
consent_server_notice_sent: null
consent_version: null
creation_ts: 1573653091
deactivated: 0
displayname: Test User 1
external_ids: []
is_guest: 0
name: '@testuser1:peek-a-boo.at'
password_hash: $secret
shadow_banned: null
threepids: []
user_type: bot

User account settings to be modified:
display_name: TESTUSER1
Are you sure you want to modify user? (y/N): y
admin: 0
appservice_id: null
avatar_url: null
consent_server_notice_sent: null
consent_version: null
creation_ts: 1573653091
deactivated: 0
displayname: TESTUSER1
external_ids: []
is_guest: 0
name: '@testuser1:peek-a-boo.at'
password_hash: $secret
shadow_banned: null
threepids: []
user_type: null
```